src/lib.rs

```diff
@@ -220,6 +238,11 @@ mod builders;
 mod errors;
 mod log_impl;
 
+/// Logger implementations for different output targets
+pub mod logger {
+    pub use crate::log_impl::{Panic, Reopen, Sender, Stderr, Stdout, Writer};
```

Comment:
If we expose these directly, I think we should _at least_ make all the fields of these structs private - they're public to the crate right now, but exposing them to the world would lock us in on these particular structures, and that's a pain when updating code.

If we get rid of the configuration altogether, or rather, merge `config` and `log_impl`, the `loggers` name you have here would make a good new name for the module file as well.

Reply:
Totally agree. I haven't decided on what API these will have, but probably simple a few constructors and `From` implementations. My goal is indeed to move them to an `logger` (or `loggers` module), but I'm using the export trick right now to keep the diff more manageable.